src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -1794,13 +1801,15 @@ fn prepare_enum_metadata(
                 .as_generator()
                 .variant_range(enum_def_id, cx.tcx)
                 .map(|variant_index| {
-                    let name = SmallCStr::new(&substs.as_generator().variant_name(variant_index));
+                    let name = substs.as_generator().variant_name(variant_index);
```

Comment:
Same as above, does `DIBuilderCreateEnumerator` take ownership of the name pointer? If so, this is a use-after-free when `variant_name` returns the owned variant of a `Cow`. Why did you remove the `SmallCStr` here?